packages/backend-test-utils/src/next/services/mockServices.ts

```diff
@@ -57,6 +58,44 @@ function simpleFactory<
   })) as (...options: TOptions) => ServiceFactory<TService, any>;
 }
 
+/** @public */
+export type ServiceMock<TService> = {
+  factory: ServiceFactory<TService>;
+} & {
+  [Key in keyof TService]: TService[Key] extends (
+    this: infer This,
+    ...args: infer Args
+  ) => infer Return
+    ? TService[Key] & jest.MockInstance<Return, Args, This>
+    : TService[Key];
+};
+
+/** @internal */
+function simpleMock<TService>(
+  ref: ServiceRef<TService, any>,
+  mockFactory: () => jest.Mocked<TService>,
+): (partialImpl?: Partial<TService>) => ServiceMock<TService> {
+  return partialImpl => {
+    const mock = mockFactory();
+    if (partialImpl) {
+      for (const [key, impl] of Object.entries(partialImpl)) {
+        if (typeof impl === 'function') {
+          (mock as any)[key].mockImplementation(impl);
```

Comment:
This is a really neat idea. Good job coming up with it. Simple and effective to shunt a little of the responsibility onto the original service maker to just make a little object of jest fns

MAYBE it's less clear that you missed making an implementation of a certain method in your test, when there's silently a jest.fn there rather than getting a TypeError because you accessed a nonexisting field - but not even sure about that.